starknet_py/tests/e2e/docs/guide/test_cairo1_contract.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from starknet_py.net.client_models import CasmClass
+from starknet_py.net.udc_deployer.deployer import _get_random_salt
+from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
+
+
+@pytest.mark.asyncio
+async def test_cairo1_contract(
+    account, sierra_minimal_compiled_contract_and_class_hash, gateway_client
+):
+    # pylint: disable=import-outside-toplevel, too-many-locals
+    (
+        contract_compiled,
+        compiled_class_hash,
+    ) = sierra_minimal_compiled_contract_and_class_hash
+
+    # docs: start
+    from starknet_py.hash.casm_class_hash import compute_casm_class_hash
+    from starknet_py.net.schemas.gateway import CasmClassSchema
+
+    # contract_compiled is the output of the starknet-sierra-compile (.casm file)
+    casm_class = CasmClassSchema().loads(contract_compiled)
+
+    # Compute Casm class hash
+    casm_class_hash = compute_casm_class_hash(casm_class)
+    # docs: end
+
+    assert casm_class_hash == compiled_class_hash
+
+    # docs: start
+
+    # Create Declare v2 transaction
+    declare_v2_transaction = await account.sign_declare_v2_transaction(
+        compiled_contract=contract_compiled,
```

Comment:
users probably don't know what `compiled_contract` means here

Reply:
There is a comment explaining that a few lines above